app/models/collection.rb

```diff
@@ -335,6 +335,29 @@ def all_approved_bookmarks_count
     count
   end
 
+  # Return the count of all bookmarkable items (works, series, external works)
+  # that are in this collection (or any of its children) and visible to
+  # the current user. Excludes bookmarks of deleted works/series.
+  def all_bookmarked_items_count
+    # The set of all bookmarks in this collection and its children.
+    # Note that "approved_by_collection" forces the bookmarks to be approved
+    # both by the collection AND by the user.
+    bookmarks = Bookmark.is_public.joins(:collection_items).
+      merge(CollectionItem.approved_by_collection).
+      where(collection_items: { collection_id: children.ids + [id] })
+
+    logged_in = User.current_user.present?
```

Comment:
Are we not concerned about admins here?

Reply:
The `User.current_user` variable contains either a `User`, an `Admin`, or `nil`, so this does actually handle admins. (It's the `ApplicationController.current_user` variable that only contains `User`s or `nil`.)

Reply:
That's not confusing... but it's a pre-existing behaviour so let's ignore it and move swiftly on 👍

Reply:
I've been thinking about trying to change it for months. I hate how easy it is to mix it up. (But I agree that now is not the time.)